utils/storage/storage.js

```diff
@@ -248,13 +251,81 @@ class Storage {
     );
   }
 
+  /**
+   * Function to fetch anonymousId from external source
+   * @param {string} key source of the anonymousId
+   * @returns string
+   */
+  fetchExternalAnonymousId(source) {
+    let anonId;
+    const key = source.toLowerCase();
+    if (!Object.keys(anonymousIdKeyMap).includes(key)) {
+      return anonId;
+    }
+    switch (key) {
+      case "segment":
+        /**
+         * First check the local storage for anonymousId
+         * Ref: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#identify
+         */
+        if (Store.enabled) {
+          anonId = Store.get(anonymousIdKeyMap[key]);
+        }
+        // If anonymousId is not present in local storage and check cookie support exists
+        // fetch it from cookie
+        if (!anonId && Cookie.IsCookieSupported()) {
+          anonId = Cookie.get(anonymousIdKeyMap[key]);
+        }
+        return anonId;
+
+      default:
+        return anonId;
+    }
+  }
+
   /**
    * get stored anonymous id
+   *
+   * Use cases:
+   * 1. getAnonymousId() ->  anonymousIdOptions is undefined this function will return rl_anonymous_id
+   * if present otherwise undefined
+   *
+   * 2. getAnonymousId(anonymousIdOptions) -> In case anonymousIdOptions is present this function will check
+   * if rl_anonymous_id is present then it will return that
+   *
+   * otherwise it will validate the anonymousIdOptions and try to fetch the anonymous Id from the provided source.
+   * Finally if no anonymous Id is present in the source it will return undefined.
+   *
+   * anonymousIdOptions example:
+   *  {
+        autoCapture: {
+          enabled: true,
+          source: "segment",
+        },
+      }
+   *
    */
-  getAnonymousId() {
-    return this.parse(
+  getAnonymousId(anonymousIdOptions) {
+    // fetch the rl_anonymous_id from storage
+    const rlAnonymousId = this.parse(
       this.decryptValue(this.storage.get(defaults.user_storage_anonymousId))
     );
+    if (rlAnonymousId) {
+      return rlAnonymousId; // return if already present
```

Comment:
Let's add a rationale for this function's code flow.

Reply:
You mean the function or the if block?

Reply:
The function.

Reply:
Already added use cases what else could be added?

Reply:
OBE with this suggestion,
https://github.com/rudderlabs/rudder-sdk-js/pull/497/files#r856312675.
Hence, resolving this conversation.